src/NServiceBus.Core/Routing/DistributionStrategy.cs

```diff
@@ -24,6 +27,14 @@ protected DistributionStrategy(string endpoint, DistributionStrategyScope scope)
         public abstract string SelectReceiver(string[] receiverAddresses);
 
         /// <summary>
+        /// Selects a destination instance for a message from all known addresses of a logical endpoint.
+        /// </summary>
+        public virtual Task<string> SelectReceiver(string[] receiverAddresses, IOutgoingContext outgoingContext)
```

Comment:
interesting thought. Is that something we should consider more often in order to allow evolution of APIs via non-breaking opt-in mechanism?